key.go

```diff
@@ -38,6 +38,21 @@ var KeyTypes = []int{
 	Secp256k1,
 }
 
+type PubKeyUnmarshaller = func(data []byte) (PubKey, error)
+type PrivKeyUnmarshaller = func(data []byte) (PrivKey, error)
```

Comment:
Nice PR! Is there a reason these are being declared as type aliases? Just ran tests with `PubKeyUnmarshaller` and `PrivKeyUnmarshaller` declared as concrete types and everything still passes.

Reply:
Good point, didn't notice the nuance! 
Perhaps we should make them concrete types, I can open a new PR for this.

Reply:
No particular reason for this.  Creating a new PR to fix it since @vyzo didn't seem to get to it yet.